### PR TITLE
fix: deposit max-fees must not exceed the deposit amount

### PR DESCRIPTION
**Pull request by djordon**
## Description

Closes https://github.com/stacks-network/sbtc/issues/932

## Changes

* Make sure we filter on the "real max-fee", which is the minimum of their max-fee and the deposit amount.
* Update the `SOLO_DEPOSIT_TX_VSIZE` to be based off of the most expensive recipient. This ensures that filtering based off of fees for deposits are conservative.

## Testing Information

This still needs a test or two.

## Checklist:

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: signer/src/bitcoin/utxo.rs [hunk 1]
```diff
@@ -54,10 +54,11 @@ const DEFAULT_INCREMENTAL_RELAY_FEE_RATE: f64 =
...
 /// This constant represents the virtual size (in vBytes) of a BTC
-/// transaction that includes two inputs and one output. The inputs
-/// consist of the signers' input UTXO and a UTXO for a deposit request.
-/// The output is the signers' new UTXO.
-const SOLO_DEPOSIT_TX_VSIZE: f64 = 234.0;
+/// transaction that includes two inputs and one output. The inputs consist
+/// of the signers' input UTXO and a UTXO for a deposit request. The output
+/// is the signers' new UTXO. The deposit request is such that the sweep
+/// transaction has the largest size of solo deposit sweep transactions.
+const SOLO_DEPOSIT_TX_VSIZE: f64 = 267.0;
```
Comment (Jiloc) [inline signer/src/bitcoin/utxo.rs:61]: sanity check and...personal curiosity: how is this value computed? Just asking to better understand the context

**Review by Jiloc — COMMENTED** [review]
Just a personal question. The implementation looks good

File: signer/src/bitcoin/utxo.rs (hunk 1 above)
Comment (djordon) [reply signer/src/bitcoin/utxo.rs:61]: We have a "test", `create_deposit_only_tx`, that creates a sweep transaction moving only one deposit and zero withdrawals that outputs the size of the sweep transaction after it has been signed. This gives us the maximum size of a sweep transaction with only one deposit. Adding additional deposits decreases the costs, and adding additional withdrawals used to decrease the cost as well. 🤔 But now maybe not, I need to check again.

**Review by Jiloc — APPROVED** [review]

File: signer/src/bitcoin/utxo.rs (hunk 1 above)
Comment (djordon) [reply signer/src/bitcoin/utxo.rs:61]: I just checked and costs decrease with an added withdrawal.

**Pull request merged**